brave/src/main/java/brave/handler/MutableSpan.java

```diff
@@ -74,113 +83,204 @@
   ArrayList<Object> annotations;
   Throwable error;
 
+  /** @since 5.4 */
   public MutableSpan() {
-    // this cheats because it will not need to grow unless there are more than 5 tags
-    tags = new ArrayList<>();
-    // lazy initialize annotations
-  }
-
-  /** Returns true if there was no data added. Usually this indicates an instrumentation bug. */
-  public boolean isEmpty() {
-    return kind == null
-      && !shared
-      && startTimestamp == 0L
-      && finishTimestamp == 0L
-      && name == null
-      && localServiceName == null
-      && localIp == null
-      && remoteServiceName == null
-      && remoteIp == null
-      && localPort == 0
-      && remotePort == 0
-      && tags.isEmpty()
-      && annotations == null
-      && error == null;
-  }
-
-  /** Returns the {@link brave.Span#name(String) span name} or null */
+  }
+
+  /** @since 5.12 */
+  public MutableSpan(MutableSpan toCopy) {
```

Comment:
If not mistaken, this has been added mainly to support `defaultSpan`. Since `defaultSpan` only has three fields specified, it seems like it might be a net-loss to copy all these other default fields when creating a span.

Reply:
not exactly but close. empty is checked, so we have to cycle through all state anyway. seems higher leverage to implement equals vs do maintenance on 3 fields that we expect to be the comparison state, yet could drift later for any reason (ex if we implement default tags with MutableSpan later)